src/Tickets/Commerce/Gateways/Square/Webhooks.php

```diff
@@ -265,6 +265,26 @@ public function verify_signature( $received_secret_key ) {
 		return wp_check_password( $unhashed_key, $received_secret_key );
 	}
 
+	/**
+	 * Verify whodat signature.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $signature The signature from the request header.
+	 * @param string $body      The raw request body.
+	 *
+	 * @return bool Whether the signature is valid.
+	 */
+	public function verify_whodat_signature( $received_secret_key ) {
+		$signature = tribe( Merchant::class )->get_whodat_signature();
+
+		if ( empty( $signature ) ) {
+			return false;
+		}
+
+		return $signature === $received_secret_key;
```

Comment:
I'd like to see a runtime evaluation of the signature. The alternative approach is a good idea, which i can support.

Simply storing once the signature though and just evaluating that they are equal after that is not what i would choose.

A runtime evaluation which would also involve the merchant id would be great. (So that if the merchant id changes in the local wp site but an event arrives for the old merchant id we had stored in whoDat, the eval should fail)